Library/Homebrew/cmd/tap.rb

```diff
@@ -42,59 +41,17 @@ def install_tap user, repo
     true
   end
 
-  def link_tap_formula(paths, warn_about_conflicts=true)
-    ignores = (HOMEBREW_LIBRARY/"Formula/.gitignore").read.split rescue []
-    tapped = 0
+  # migrate from symlink based tap structure to directories based structure
+  def migrate_taps(options={:autoskip => true})
+    ignore = HOMEBREW_LIBRARY/"Formula/.gitignore"
+    return if options[:autoskip] && !ignore.exist?
```

Comment:
I use `gitignore` file to determine whether we have migrated or not. It `gitingore` file doesn't exist, we can skip it. But when doing `brew tap --repair`, I'm thinking it should force the migration and we should scan the symlink files regardless of the `gitignore` file.